### PR TITLE
Add Specs: movement testnet bardock

### PR DESCRIPTION
**Pull request by diszell2008**
Hi Lava Foundation,
This spec use for movement bardock testnet chain, will update spec when movement mainnet run.

Discord : diszell2008

### CODE REVIEW

**Comment by ignacio-rsg** [comment]
1. Please add a description for the spec. You can use the same one provided for the mainnet specification if applicable.
2. Test the spec locally using [this script](https://github.com/lavanet/lava/blob/dc487d4a0d1fef81195710e23c363a1f2ec5b21f/scripts/test_spec_full.sh).
3. Once tested, please upload the logs to the PR for review and the endpoints used for testing(if public)

**Comment by diszell2008** [comment]
Log tested specs with local script.
```
Processing index: MOVEMENTT
gas estimate: 319888
code: 0
codespace: ""
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: '[]'
timestamp: ""
tx: null
txhash: 6C9B7B0250A04E35A34E748C19182AAD63A0F9A5820E4A70D103D07F0328CF09
waiting for next block "24"
finished waiting at block "25"
Waiting for the next epoch for the changes to be active
finished waiting
[+]generated provider config: /home/ubuntu/lava/scripts/../testutil/debugging/logs/provider.yml
endpoints:
    - api-interface: rest
      chain-id: MOVEMENTT
      network-address:
        address: "127.0.0.1:2220"
      node-urls:
        - url: http://15.235.202.242:30731/v1
[Test Setup] starting provider
[+]generated consumer config: /home/ubuntu/lava/scripts/../testutil/debugging/logs/consumer.yml
endpoints:
    - chain-id: MOVEMENTT
      api-interface: rest
      network-address: 0.0.0.0:3001
[+] letting providers start and running health check then running command with flags:  http://0.0.0.0:3001 MOVEMENTT rest
Nov 14 08:13:04 INF RPCConsumer Test started args=http://0.0.0.0:3001;MOVEMENTT;rest
Nov 14 08:13:04 INF setting log level loglevel=debug
Nov 14 08:13:04 INF lavad Binary Version: 4.1.1-dirty
Nov 14 08:13:04 DBG batched nodeUrls by services batch="map[||:{NetworkAddress:{Address: KeyPem: CertPem: DisableTLS:false} ChainID:MOVEMENTT ApiInterface:rest Geolocation:1 NodeUrls:[http://0.0.0.0:3001]}]"
Nov 14 08:13:04 DBG chainRouter created chainProxyRouter="map[||:[{ChainProxy:0xc0016726c0 addonsSupported:map[:{}] methodsRouted:map[]}]]"
Nov 14 08:13:04 DBG Chain Tracker Updated block hashes ApiInterface=rest ChainID=MOVEMENTT blocksKept=10 blocksQueried=0 blocksQueueLen=10 latestHash=JFU= latest_block=2455 nextBlocksUpdate=2555
Nov 14 08:13:04 DBG ChainTracker fetched init data successfully
![Uploading Screenshot 2024-11-14 at 15.13.53.png…]()

```

**Comment by diszell2008** [comment]
Public LAVA RPC consumer for test : http://15.235.202.242:3001/

File: mainnet-1/specs/movementt.json [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    "proposal": {
+        "title": "Add Specs: Movement Testnet",
+        "description": "Adding new specification support for relaying Movement testnet bardock data on Lava",
+        "specs": [
+            {
+                "index": "MOVEMENTT",
+                "name": "movement testnet bardock",
+                "enabled": true,
+                "imports": [
+                    "APT1"
```
Comment (Yaroms) [inline mainnet-1/specs/movementt.json:11]: import from movement mainnet

File: mainnet-1/specs/movementt.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                        "enabled": true,
+                        "collection_data": {
+                            "api_interface": "rest",
+                            "internal_path": "",
+                            "type": "GET",
+                            "add_on": ""
+                        },
+                        "apis": [
+                            {
+                                "name": "/transactions/wait_by_hash/{txn_hash}",
```
Comment (Yaroms) [inline mainnet-1/specs/movementt.json:35]: if you import movement spec you dont need this one

File: mainnet-1/specs/movementt.json [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                                    "stateful": 0
+                                },
+                                "extra_compute_units": 0
+                            }
+                        ],
+                        "headers": [
+                            {
+                                "name": "x-aptos-gas-used",
+                                "kind": "pass_reply",
+                                "function_tag": "DISABLED"
```
Comment (Yaroms) [inline mainnet-1/specs/movementt.json:57]: why is this header disabled?

Comment (diszell2008) [reply mainnet-1/specs/movementt.json:57]: we need enable all ? 
Screenshot 2024-11-14 at 20 01 50

File: mainnet-1/specs/movementt.json [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                                    {
+                                        "expected_value": "250"
+                                    }
+                                ]
+                            },
+                            {
+                                "name": "pruning",
+                                "values": [
+                                    {
+                                        "latest_distance": 2400
```
Comment (Yaroms) [inline mainnet-1/specs/movementt.json:75]: this should be the number of blocks in 24H

File: mainnet-1/specs/movementt.json (hunk 2 above)
Comment (diszell2008) [reply mainnet-1/specs/movementt.json:35]: yes.

File: mainnet-1/specs/movementt.json (hunk 3 above)
Comment (Yaroms) [reply mainnet-1/specs/movementt.json]: if you inherit it from aptos than you dont need to do it again

File: mainnet-1/specs/movement.json [hunk 5]
```diff
@@ -375,38 +375,38 @@
                         ],
                         "headers": [
                             {
-                                "name": "x-aptos-ledger-version",
-                                "kind": "pass_reply"
+                              "name": "x-aptos-ledger-version",
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:378]: no need to specify all of those, we get them from aptos

File: mainnet-1/specs/movement.json [hunk 6]
```diff
@@ -472,12 +475,17 @@
                                 },
                                 "values": [
                                     {
-                                        "latest_distance": 432000
-                                    },
-                                    {
+                                        "extension": "",
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:478]: you dont need to add fields that are with default values (empty strings etc..)
also if the verification is the same as aptos you dont need to specify the parse directive, it will take it from there

File: mainnet-1/specs/movement.json [hunk 7]
```diff
@@ -669,60 +677,104 @@
...
+                        "extensions": [
+                            {
+                                "rule": {
+                                    "block": 1
+                                }
+                            }
+                        ]
+                    },
+                    {
+                        "enabled": true,
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:708]: this api collection does not add anything, delete it

File: mainnet-1/specs/movement.json [hunk 8]
```diff
@@ -669,60 +677,104 @@
...
-                                        "latest_distance": 432000
-                                    },
+                                        "latest_distance": 2400
+                                    }
+                                ]
+                            }
+                        ],
+                        "extensions": [
+                            {
+                                "rule": {
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:701]: you need to specify that this is archive extension

File: mainnet-1/specs/movement.json [hunk 9]
```diff
@@ -669,60 +677,104 @@
...
                                 ]
                             }
                         ],
                         "extensions": [
                             {
-                                "name": "archive",
-                                "cu_multiplier": 5,
                                 "rule": {
-                                    "block": 427500
+                                    "block": 1
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:777]: specify the name of the extension, archive

File: mainnet-1/specs/movement.json [hunk 10]
```diff
@@ -618,7 +626,7 @@
             {
                 "index": "MOVEMENT",
                 "name": "movement mainnet",
-                "enabled": true,
+                "enabled": false,
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:629]: Why is this disabled

File: mainnet-1/specs/movement.json [hunk 11]
```diff
@@ -669,50 +677,21 @@
...
-                                    "api_name": "/"
-                                },
                                 "values": [
                                     {
-                                        "latest_distance": 432000
-                                    },
-                                    {
-                                        "extension": "archive",
-                                        "expected_value": "0"
+                                        "latest_distance": 2400
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:694]: Why did you remove the archive pruning?

File: mainnet-1/specs/movement.json [hunk 12]
```diff
@@ -722,32 +701,82 @@
...
+                            {
+                                "name": "chain-id",
+                                "values": [
+                                    {
+                                        "expected_value": "250"
+                                    }
+                                ]
+                            },
+                            {
+                                "name": "pruning",
```
Comment (Yaroms) [inline mainnet-1/specs/movement.json:755]: What archive verification?
